src/screens/MainScreen/index.js

```diff
@@ -28,11 +32,34 @@ const MainScreen = () => {
             provider={PROVIDER_GOOGLE}
             style={styles.map}
             region={{
-              latitude: 37.78825,
-              longitude: -122.4324,
+              latitude: userHasLocation ? latitude : 37.78825,
```

Comment:
What if you save the default coordinates values in constants and use them from there?